packages/cli/src/plugins/resource/plugin-optimization-mpa.js

```diff
@@ -33,7 +33,7 @@ class OptimizationMPAResource extends ResourceInterface {
   }
 
   async shouldOptimize(url) {
-    return Promise.resolve(path.extname(url) === '.html' && this.compilation.config.mode === 'mpa');
+    return Promise.resolve(url !== '404.html' && path.extname(url) === '.html' && this.compilation.config.mode === 'mpa');
```

Comment:
If we ever get around to adding a 500 page or something, we might want a better system for identifying this "special" pages